src/main/java/org/pac4j/vertx/auth/Pac4jUser.java

```diff
@@ -82,8 +82,8 @@ public int readFromBuffer(int pos, Buffer buffer) {
         final byte[] jsonBytes = buffer.getBytes(pos, pos + jsonByteCount);
         pos += jsonByteCount;
         final String json = new String(jsonBytes, StandardCharsets.UTF_8);
-        final UserProfile userProfile = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
-        setUserProfile(userProfile);
+        final UserProfile userProfileLocal = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
```

Comment:
I don't see why that's a problem. Java will handle the scoping, so no need for a variable named "...Local". To me this feels rather like an over-zealous static analyser. We can even coalesce lines 85 and 86 so there's no explicit local variable for it. I'm certainly not in favour of using "Local" to distinguish, we're going to need a more appropriate name IMO.